### DISCUSSION TITLE
Single Product Page - Double buttons for updating quantity

### DISCUSSION THREAD

**Discussion by antforager** [bootscore/bootscore#486] [Q&A] [answered]
I'm currently updating my child theme from 5.2.0.0 to 5.2.3.4. My bootscore version is on 5.2.3.4.

On the single product page, there are duplicate quantity buttons for plus and minus. 

In /bootscore-main/woocommerce/woocommerce-functions.php, there is a section that adds the plus and minus buttons:

`
function bs_quantity_minus_button() {
  echo '<button type="button" class="minus input-group-text" >-</button>';
}
`

Somewhere, there is duplicate code adding this html (got this from view source):
`<button type="button" class="input-group-text qty_button minus">-</button>`

and

`<button type="button" class="input-group-text qty_button plus">+</button>`

I don't know where this extra code is coming from. I'm not overriding it in my child theme. In v. 5.2 of my child theme this doesn't happen. 

Any ideas on how I should troubleshoot this?

Thanks,
Cindy

**Comment by crftwrk** [comment]
Do you have a plugin installed which adds them as well?

**Reply by antforager** [reply]
I have searched high and low for this -- throughout the plugin files and the only thing that is adding an action to `woocommerce_before_quantity_input_field` is the bootscore wootcommerce-functions.php code I mentioned above. I disabled a bunch of plugins that could be affecting the page and it still showed the extra buttons. Then I decided to go crazy and remove the +/- code from bootscore's wootcommerce-functions.php class. I saved the file and it did absolutely nothing. So, it's been caching stuff? So, how do I know if the plugins that were disabled actually did anything? I need to have some kind of event trace so I can see what happened and when. Not sure if that exists.

**Reply by crftwrk** [reply]
We refactored the quantity buttons in February https://github.com/bootscore/bootscore/pull/405.

> So, it's been caching stuff?

Maybe, do you have a caching plugin installed?

Please do this:

- Disable ALL plugins (except WooCommerce)
- Try to remove function as you did in your comment
- Do you have a copy of `quantity-input.php` in your child? If so, remove it.
- Purge cache

As mentioned in my previous comment, I'm pretty sure that there is a conflict with a plugin. So, let's dig into it. BTW, a link to your project is always very very helpful ;-)

**Reply by antforager** [reply]
I disabled all plugins except for WooCommerce. sampler/

That section of code is removed (the whole thing, even the action).

I purged cache.

Still there...

**Reply by crftwrk** [reply]
Well, this is a mystery...

We had a overridden `quantity-input.php` with this buttons and they used `input-group-text qty_button minus` class. We deleted this template and created a function instead https://github.com/bootscore/bootscore/pull/405. New buttons use `minus input-group-text` class.

On your site, both button classes (old, overridden template and new, function) exists. There are only two possible explanations:

1. You have a copy of the old `quantity-input.php` in your child.
2. There is a really really hard server-side caching.

While point 1 seems to be unrealistic, I guess it's point 2. As you mentioned in your comment, you removed the code for the buttons in main-theme and this had no effect. If PHP code is removed and doesn't make anything in the frontend, then it must be a server-side caching. 

This can be done by several plugins like W3 Total Cache etc., the `.htaccess` or a setting in your hosters backend. Try to move the site to a different server and/or check the server settings.

**Reply by antforager** [reply]
Yesterday, I signed up for another wordpress account. I'm going to do exactly what you suggest and hopefully I can copy the site over.

I did have a caching plugin and I'll bet this is where it's coming from. Last week, I added an "Out of Stock" lable on the category page. It was bright red so I changed it to be a muted shade of red and it looked pink. I changed it back to the darker red and it took almost a whole week to reflect. I thought I was stuck with pink. 

Cindy

**Reply by crftwrk** [reply, marked as answer]
Let me share just one more idea...

> I did have a caching plugin and I'll bet this is where it's coming from.

Is this plugin uninstalled or deactivated? I don't know which plugin you used, but almost all of the optimising plugins write something to your `.htaccess` file. If you just deactivated or deleted the plugin, then it's possible that the plugins code remains in your `.htaccess` file. And if that code says to the server that he should cache, he caches. No matter if plugin is installed or not.

So, you can open your `.htaccess` and have a look if there is some remaining code from the plugin. But much better, install and activate plugin again and purge cache in backend.

BTW, you can easily transfer any site to any server. There are many plugins out there. I use [Akeeba Backup](https://www.akeeba.com/products/akeeba-backup-wordpress.html) for the backups and [Akeeba Kickstart](https://www.akeeba.com/download/akeeba-kickstart.html) to restore the backup whereever I want. Super simple.

Edit: If all above not helps, hide the old buttons roughly with CSS ;-)

```css
.qty_button.minus, 
.qty_button.plus {
  display: none;
}
```

**Reply by antforager** [reply]
I don't know what to do about this caching problem. The colors and images are still the "bootscore" version even though I changed them after uploading the theme. My .htaccess file looks ok.

**Comment by antforager** [comment]
I decided to stop pussyfooting around with this so I upgraded the theme in production. It works fine with no double buttons.
